servo/api.py

```diff
@@ -285,6 +295,104 @@ async def _post_event(self, event: Events, param) -> Union[CommandResponse, Stat
                 self.logger.trace(_redacted_to_curl(error.request))
                 raise
 
+    async def _get_diagnostics(self) -> Diagnostics:
+
+        async with aiofiles.open(logs_path, 'r') as log_file:
+            log_data_lines = await log_file.readlines()
+
+        # Format and strip emoji from logs :(
+        log_dict = {}
+        for line in log_data_lines[:4000]: # Limit to 4000 lines for <1 mb assets limit
+            time, msg = line.split('|', 1)
+            log_dict[time.strip()] = msg.strip().encode("ascii", "ignore").decode()
+
+        export_options = dict(exclude_unset=True, exclude_none=True)
+
+        # NOTE: Round-trip through JSON to produce primitives
+        config_dict = self.servo.config.json(**export_options)
+        config_data = json.loads(config_dict)
+
+        return Diagnostics(configmap=config_data, logs=log_dict)
+
+
+    async def _diagnostics_request(self) -> DiagnosticStates:
```

Comment:
There's a lot of repetition between the methods `_diagnostics_request`, `_post_diagnostics`, and `_reset_diagnostics`. Please refactor the common code into a helper method in servo/telemetry.py

Reply:
Good call